src/grids/linear_grid.jl

```diff
@@ -1,217 +1,218 @@
-    points = δ .* collect(indices) .+ xmin
-    points[end] = xmax
-    
-    new{T}(xmin, xmax, len, points, δ, LinearGridOps(points, complex_op; δ))
-  end
-  #
-  LinearGrid(::Type{T}, args...; kwargs...) where T = LinearGrid(args...; T, kwargs...)
+
+        indices = 0:(len-1)
+
+        d = 2abs(grid_min) / len
+
+        # Symmetric grid around 0, note that the last point is not included
+        points = d .* (collect(indices) .- len ÷ 2)
+
+        new{T}(grid_min, len, points, d, LinearGridOps(points, complex_op; d))
+    end
+    #
+    LinearGrid(::Type{T}, args...; kwargs...) where {T} = LinearGrid(args...; T, kwargs...)
 end
 
 """
 Apply FFT of grid to input
 """
-function fft!(grid::LinearGrid, out::AbstractVector, u::AbstractVector)
-  (; padded, ft, ft⁻¹, ft_shift) = grid.op
-  
-  assert_applicable(ft, padded, out)
-  unsafe_execute!(ft, _zeropad!(padded, u), out)
-  @. out *= ft_shift
-  
-  return out
+function ft!(grid::LinearGrid, out::AbstractVector, u::AbstractVector)
+    return fftshift(bfft(u)) .* grid.op.ft_arr[1:length(u)] * grid.d
 end
 
-fft(grid::LinearGrid, u::AbstractVector) = fft!(grid, zero(grid.op.ft_work1), u)
+ft(grid::LinearGrid, u::AbstractVector) = ft!(grid, zero(u), copy(u))
 
 """
 Internal function that works on padded arrays
 """
-function _ifft!(grid::LinearGrid, out::AbstractVector, u::AbstractVector)
-  (; ft_work1, ft⁻¹, ft⁻¹_shift) = grid.op
-  
-  assert_applicable(ft⁻¹.p, ft_work1, out)
-  _zeropad!(ft_work1, u; fillzero=true)
-  @. ft_work1 *= ft⁻¹_shift
-  unsafe_execute!(ft⁻¹.p, ft_work1, out)
-  
-  return out
+function ift!(grid::LinearGrid, out::AbstractVector, u::AbstractVector)
+    return fftshift(fft(u)) .* grid.op.ft_arr[1:length(u)] * grid.d / 2π
 end
 
 """
 Apply inverse FFT of grid to input
 """
-ifft!(grid::LinearGrid, out::AbstractVector, u::AbstractVector
-  ) = out .= @views _ifft!(grid, grid.op.ft⁻¹_work, u)[1:grid.len]
 
-ifft(grid::LinearGrid, u::AbstractVector
-  ) = ifft!(grid, zeros(eltype(grid.op.ft⁻¹_work), grid.len), u)
+ift(grid::LinearGrid, u::AbstractVector) = ift!(grid, zero(u), copy(u))
 
 """
 Internal function that works on padded arrays
 """
 function _convolution!(grid::LinearGrid, out::AbstractVector,
-    u::AbstractVector,v::AbstractVector)
-  (; padded, ft_work1, ft_work2, ft, ft⁻¹, conv_shift) = grid.op
-  
-  assert_applicable(ft⁻¹.p, ft_work2, out)
-  unsafe_execute!(ft, _zeropad!(padded, u), ft_work1)
-  unsafe_execute!(ft, _zeropad!(padded, v), ft_work2)
-  @. ft_work2 *= ft_work1 * conv_shift
-  unsafe_execute!(ft⁻¹.p, ft_work2, out)
-  
-  return out
+    u::AbstractVector, v::AbstractVector)
+    (; padded, ft_work1, ft_work2, ft_p, bft_p, ft_arr) = grid.op
+
+    assert_applicable(bft_p, ft_work2, out)
+    unsafe_execute!(ft_p, _zeropad!(padded, u), ft_work1)
+    unsafe_execute!(ft_p, _zeropad!(padded, v), ft_work2)
+    @. ft_work2 *= ft_work1 * ft_arr
+
+    unsafe_execute!(bft_p, ft_work2, out)
+
+    out .*= grid.d / length(padded)
```

Comment:
Performance wise it should be much better to multiply this into ft_work2 before the FT, I don't know if this is bad for the data though. Should not be, but who knows.